plasmapy/simulation/particle_tracker.py

```diff
@@ -494,14 +494,99 @@ def setup_adaptive_time_step(
+        sets the `_is_synchronized_time_step` and `_is_adaptive_time_step` attributes.
+        """
+
+        if dt_range is not None and not self._is_adaptive_time_step:
+            raise ValueError(
+                "Specifying a time step range is only possible for an adaptive time step."
+            )
+
+        # Load and validate inputs
+        field_weightings = ["volume averaged", "nearest neighbor"]
+        if field_weighting in field_weightings:
+            self.field_weighting = field_weighting
+        else:
+            raise ValueError(
+                f"{field_weighting} is not a valid option for ",
+                "field_weighting. Valid choices are",
+                f"{field_weightings}",
+            )
+
+    def _validate_grids(self, additional_required_quantities):
+        """Add required quantities to grid objects.
+
+        Grids lacking the required quantities will be filled with zeros.
+        """
+
+        # Some quantities are necessary for the particle tracker to function regardless of other configurations
+        required_quantities = {"E_x", "E_y", "E_z", "B_x", "B_y", "B_z"}
+
+        if additional_required_quantities is not None:
+            # Add additional required quantities based off simulation configuration
+            required_quantities.update(additional_required_quantities)
+
+        for grid in self.grids:
+            grid.require_quantities(required_quantities, replace_with_zeros=True)
+
+            for rq in required_quantities:
+                # Check that there are no infinite values
+                if not np.isfinite(grid[rq].value).all():
+                    raise ValueError(
+                        f"Input arrays must be finite: {rq} contains "
+                        "either NaN or infinite values."
+                    )
+
+                # Check that the max values on the edges of the arrays are
+                # small relative to the maximum values on that grid
+                #
+                # Array must be dimensionless to re-assemble it into an array
+                # of max values like this
+                arr = np.abs(grid[rq]).value
+                edge_max = np.max(
+                    np.array(
+                        [
+                            np.max(a)
+                            for a in (
+                                arr[0, :, :],
+                                arr[-1, :, :],
+                                arr[:, 0, :],
+                                arr[:, -1, :],
+                                arr[:, :, 0],
+                                arr[:, :, -1],
+                            )
+                        ]
+                    )
+                )
+
+                if edge_max > 1e-3 * np.max(arr):
+                    unit = grid.recognized_quantities[rq].unit
+                    warnings.warn(
+                        "Fields should go to zero at edges of grid to avoid "
```

Comment:
Let's change "Fields" -> "Quantities" throughout this message, since we'll soon be adding non-field quantities like mass density